### PR TITLE
Azure subnet

### PR DESCRIPTION
**Pull request by JunqiZhang0**
Added Azure subnet creation and destruction

### CODE REVIEW

File: docs/source/azure.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ example: docs/source/example/workspaces/azure/azure.key
+
+
+
+.. include:: credentials/aws.rst
+
+Environment Variables
+~~~~~~~~~~~~~~~~~~~~~
+
+LinchPin honors the AWS environment variables
```
Comment (14rcole) [inline docs/source/azure.rst]: AWS?

Comment (JunqiZhang0) [reply docs/source/azure.rst]: oh Thank you! will change it

File: linchpin/provision/roles/azure/tasks/provision_azure_vm.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+---
+- name: "Set default res_grp_vars when not defined"
+  set_fact:
+    res_grp_vars:
+      - name: "{{ res_def['res_name'] | default(res_def['vm_name']) }}"
+        resource_group_name: "{{ res_grp_name  }}"
+
+# - name: Update Instance Tags
```
Comment (samvarankashyap) [inline linchpin/provision/roles/azure/tasks/provision_azure_vm.yml]: @JunqiZhang0  do we need these comments ??

Comment (JunqiZhang0) [reply linchpin/provision/roles/azure/tasks/provision_azure_vm.yml]: No

**Comment by JunqiZhang0** [comment]
[merge]

**Pull request merged**